File: core/src/main/java/io/crossroads/jna/xs_msg_t.java
package io.crossroads.jna;

import com.sun.jna.*;

public class xs_msg_t extends Structure {
	public Pointer content;
	public byte flags;
	public byte vsm_size;
	public byte[] vsm_data = new byte[XS_MAX_VSM_SIZE];
	private static final int XS_MAX_VSM_SIZE = CrossroadsIO.XS_MAX_VSM_SIZE();
}

File: jna/xs_msg_t.scala
package io.crossroads.jna

import com.sun.jna.Structure
import com.sun.jna.Pointer

class xs_msg_t extends Structure {
	var content: Pointer = _
	var flags: Byte = _
	var vsm_size: Byte = _
	var vsm_data: Array[Byte] = new Array[Byte](XS_MAX_VSM_SIZE)
	val XS_MAX_VSM_SIZE = CrossroadsIO.XS_MAX_VSM_SIZE
}